### ISSUE TITLE
rollup -c fails if rollup.config.js is a symlink

### ISSUE THREAD

**Issue by milahu** [rollup/rollup#3781]
to reproduce:
```sh
mv rollup.config.js rollup.config.js.FILE
ln -s rollup.config.js.FILE rollup.config.js
rollup -c
```

### Expected Behavior
read config via symlink

### Actual Behavior
```
[!] SyntaxError: Cannot use import statement outside a module
/home/user/src/javascript/jsdom/jsdom/rollup.config.js.file:1
import commonjs from '@rollup/plugin-commonjs';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

### env
- Rollup Version: v2.23.0
- Operating System (or Browser): linux
- Node Version (if applicable): v14.5.0

**Comment by shellscape** [comment]
Thanks for opening an issue. Citing the issue template:

>   Issues without minimal reproductions will be closed! Please provide a link to one by:
>   1. Using the REPL at https://rollupjs.org/repl/, or
>   2. Using the REPL.it reproduction template at https://repl.it/@rollup/rollup-repro
>      (allows full use of all rollup options and plugins), or
>   3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
>      These may take more time to triage than the other options.

Please add a reproduction and we'll be happy to triage further.

**Comment by milahu** [comment]
reproduction repo is here
https://github.com/milahu/rollup-bug-demo-config-symlink

symlink-ed configs are treated as commonjs files
so the es6 `import x from 'y'` fails

**Comment by lukastaegert** [comment]
Thanks for taking the time. With your reproduction, it was easy to debug and find the root cause, fix at #3783.
